clients/src/main/java/org/apache/kafka/clients/CommonClientConfigs.java

```diff
@@ -115,7 +115,9 @@
     public static final String MAX_POLL_INTERVAL_MS_DOC = "The maximum delay between invocations of poll() when using " +
             "consumer group management. This places an upper bound on the amount of time that the consumer can be idle " +
             "before fetching more records. If poll() is not called before expiration of this timeout, then the consumer " +
-            "is considered failed and the group will rebalance in order to reassign the partitions to another member. ";
+            "is considered failed and the group will rebalance in order to reassign the partitions to another member. " +
+            "For static members reaching this timeout, the expiration will be <code>max.poll.interval.ms</code> + <code>session.timeout.ms</code> " +
```

Comment:
I think the main thing to emphasize is that the group will not rebalance. How about this?

> For consumers using a non-null `group.instance.id` which reach this timeout, partitions will not be immediately reassigned. Instead, the consumer will stop sending heartbeats and partitions will be reassigned after expiration of the session timeout. This mirrors the behavior of a static consumer which has shutdown.

As an aside, do we think this behavior is intuitive? It kind of feels like we just do it so that the user can use `max.poll.interval.ms` to set a low rebalance timeout. If we had the rebalance timeout as a separate config, would we still implement this behavior?

Reply:
I do like the new comment. It's a bit tricky to say if we have two separate configs as
`max.poll.interval.ms` emphasizes more on the consumer efficiency, but we couldn't immediately stop the consumer by sending a leave group request to the broker, as we haven't implemented the handling logic on broker side. With leaveGroup changes on broker 2.4, we may actually address this problem by allowing static member to leave group explicitly too.